funcs/func_respoke_metadata.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Respoke - Web communications made easy
+ *
+ * Copyright (C) 2015, D.C.S. LLC
+ *
+ * Chad McElligott <cmcelligott@respoke.io>
+ * David M. Lee, II <dlee@respoke.io>
+ *
+ * See https://www.respoke.io for more information about
+ * Respoke. Please do not directly contact any of the
+ * maintainers of this project for assistance.
+ * Respoke offers a community forum to submit and discuss
+ * issues at http://community.respoke.io; please raise any
+ * issues there.
+ *
+ * See http://www.asterisk.org for more information about
+ * the Asterisk project. Please do not directly contact
+ * any of the maintainers of this project for assistance;
+ * the project provides a web site, mailing lists and IRC
+ * channels for your use.
+ *
+ * This program is free software, distributed under the terms of
+ * the GNU General Public License Version 2. See the LICENSE file
+ * at the top of the source tree.
+ */
+
+/*! \file
+ *
+ * \brief Retrieve data from a respoke session's metadata
+ *
+ * \author \verbatim Chad McElligott <cmcelligott@respoke.io> \endverbatim
+ * \author \verbatim David M. Lee, II <dlee@respoke.io> \endverbatim
+ *
+ * \ingroup functions
+ *
+ */
+
+/*** MODULEINFO
+	<depend>res_respoke</depend>
+	<depend>res_respoke_session</depend>
+	<support_level>extended</support_level>
+ ***/
+
+#include "asterisk.h"
+
+ASTERISK_FILE_VERSION(__FILE__, "$Revision$")
+
+#include "asterisk/app.h"
+#include "asterisk/pbx.h"
+#include "asterisk/module.h"
+#include "asterisk/channel.h"
+#include "asterisk/json.h"
+
+#include "asterisk/respoke.h"
+#include "asterisk/res_respoke_session.h"
+
+/*** DOCUMENTATION
+	<function name="RESPOKE_METADATA" language="en_US">
+		<synopsis>
+			Get a key from a respoke session's metadata
+		</synopsis>
+		<syntax>
+			<parameter name="key" required="true">
+				<para>The key in the metadata to retrieve.</para>
```

Comment:
We could probably do with more text here:

* if the key does not exist, it returns an empty string
* if the key refers to a object or array, it is stringified
* if no key is given, it renders the entire metadata object
* anything else?